transport/transport-backend-cct/src/test/java/com/google/android/datatransport/cct/CctTransportBackendTest.java

```diff
@@ -157,32 +137,55 @@ public void testCCTSuccessLoggingRequest() {
-                                .equals(
-                                    NetworkConnectionInfo.newBuilder()
-                                        .setNetworkTypeValue(activeNetworkInfo.getType())
-                                        .setMobileSubtypeValue(activeNetworkInfo.getSubtype())
-                                        .build())))
-            .andMatching(firstLogEventMatcher.test(e -> e.getEventCode() == 0))
-            .andMatching(
-                secondLogEventMatcher
-                    .test(e -> e.getEventCode() == 5)
-                    .test(e -> e.getSourceExtensionJsonProto3().equals(JSON_PAYLOAD))));
+            .withHeader("Content-Type", equalTo("application/json"))
+            .withRequestBody(matchingJsonPath("$[?(@.logRequest.size() == 1)]"))
+            .withRequestBody(
+                matchingJsonPath(
+                    String.format(
+                        "$[?(@.logRequest[0].requestTimeMs == %s)]", wallClock.getTime())))
+            .withRequestBody(
+                matchingJsonPath(
+                    String.format(
+                        "$[?(@.logRequest[0].requestUptimeMs == %s)]", uptimeClock.getTime())))
+            .withRequestBody(matchingJsonPath("$[?(@.logRequest[0].logEvent.size() == 2)]"))
+            .withRequestBody(
+                matchingJsonPath(
+                    String.format(
+                        "$[?(@.logRequest[0].logEvent[0].eventTimeMs == \"%s\")]",
+                        INITIAL_WALL_TIME)))
+            .withRequestBody(
+                matchingJsonPath(
+                    String.format(
+                        "$[?(@.logRequest[0].logEvent[0].eventUptimeMs == \"%s\")]",
+                        INITIAL_UPTIME)))
+            .withRequestBody(
+                matchingJsonPath(
+                    String.format(
+                        "$[?(@.logRequest[0].logEvent[0].sourceExtension == \"%s\")]",
+                        PAYLOAD_BYTE64)))
+            .withRequestBody(
+                matchingJsonPath(
+                    String.format(
+                        "$[?(@.logRequest[0].logEvent[0].timezoneOffsetSeconds == \"%s\")]",
+                        getTzOffset())))
+            .withRequestBody(
+                matchingJsonPath(
+                    String.format(
+                        "$[?(@.logRequest[0].logEvent[0].networkConnectionInfo.networkType == \"%s\")]",
+                        NetworkConnectionInfo.NetworkType.forNumber(activeNetworkInfo.getType()))))
+            .withRequestBody(
+                matchingJsonPath(
+                    String.format(
+                        "$[?(@.logRequest[0].logEvent[0].networkConnectionInfo.mobileSubtype == \"%s\")]",
+                        NetworkConnectionInfo.MobileSubtype.forNumber(
+                            activeNetworkInfo.getSubtype()))))
+            .withRequestBody(matchingJsonPath("$[?(@.logRequest[0].logEvent[0].eventCode == 0)]"))
```

Comment:
@vkryachko I didn't include the `eventCode` in the generated json if it wasn't set, but the test expects a `0`, so I modified the code to always set the code to 0 by default. Is this the way it's intended to work?